### PR TITLE
Gui Decimator (#352)

### PR DESCRIPTION
**Pull request by dinhlongviolin1**
Solve Issue #352 

```
import pandas as pd
import numpy as np
from src.taipy.gui import Gui
from src.taipy.gui.data.decimator import MinMaxDecimator, RDP, LTTB

df = pd.DataFrame(np.random.randint(0, 10000000, size=(30000, 1)), columns=["A"])
df["index1"] = df.index

# decimator_class = RDP(epsilon=100)
# decimator_class = RDP(n_out=300)
# decimator_class = LTTB(n_out=300)
# decimator_class = MinMaxDecimator(n_out=300)

gui = Gui(
    page=""" 
<|{df}|chart|x=index1|y=A|limit_rows|decimator=decimator_class|>
"""
)
gui.run()

```

### CODE REVIEW

File: src/taipy/gui/gui.py [hunk 1]
```diff
@@ -734,6 +734,16 @@ def _get_user_function(self, func_name: str) -> t.Union[t.Callable, str]:
             return func
         return func_name
 
+    def _get_user_class(self, class_name: str, class_type: type) -> t.Union[object, str]:
```
Comment (FredLL-Avaiga) [inline src/taipy/gui/gui.py]: _get_user_instance ?

File: src/taipy/gui/data/pandas_data_accessor.py [hunk 2]
```diff
@@ -264,14 +265,19 @@ def __get_data(  # noqa: C901
             )
         else:
             ret_payload["alldata"] = True
+            decimator = payload.get("decimator", None)
+            decimator_class = (
```
Comment (FredLL-Avaiga) [inline src/taipy/gui/data/pandas_data_accessor.py]: decimator_instance ?

**Review by FredLL-Avaiga — APPROVED** [review]
Should work :-)

File: src/taipy/gui/gui.py (hunk 1 above)
Comment (dinhlongviolin1) [reply src/taipy/gui/gui.py]: make sense! thx

File: src/taipy/gui/data/pandas_data_accessor.py (hunk 2 above)
Comment (dinhlongviolin1) [reply src/taipy/gui/data/pandas_data_accessor.py]: agree!

File: src/taipy/gui/data/utils.py [hunk 3]
```diff
@@ -12,16 +12,43 @@
...
+    def _is_applicable(self, data: t.Any, nb_rows_max: int):
+        if self.applied_threshold is None:
+            if nb_rows_max < len(data):
+                return True
+        elif self.applied_threshold < len(data):
+            return True
+        return False
+
+    @abstractmethod
+    def decimate(self, data: np.ndarray) -> np.ndarray:
```
Comment (FabienLelaquais) [inline src/taipy/gui/data/utils.py:38]: We'll need a bit more explanation at this point.
That IS the real magic of all of this.

File: src/taipy/gui/renderers/factory.py [hunk 4]
```diff
@@ -97,7 +97,7 @@ class _Factory:
                 ("on_range_change", PropertyType.function),
                 ("active", PropertyType.dynamic_boolean, True),
                 ("limit_rows", PropertyType.boolean),
```
Comment (FabienLelaquais) [inline src/taipy/gui/renderers/factory.py]: Do we still need 'limit_rows`?

**Review by FabienLelaquais — APPROVED** [review]
That looks really, really neat!

File: src/taipy/gui/renderers/factory.py (hunk 4 above)
Comment (dinhlongviolin1) [reply src/taipy/gui/renderers/factory.py]: nice catch! no we do not

File: src/taipy/gui/data/utils.py (hunk 3 above)
Comment (dinhlongviolin1) [reply src/taipy/gui/data/utils.py:38]: Can you suggest anything to add

Comment (FabienLelaquais) [reply src/taipy/gui/data/utils.py:38]: First of all, I would rename the class 'Decimator', for short. The ABC aspect is an implementation detail.
And this class should definitively get some doc. At least a skeleton, full of TODO notes.

Then, this method really is the entry point of that feature. So we will need to document it thoughtfully. 
Again, TODO entries is fine to me.

**Pull request merged**